### PR TITLE
AND #159 Implement key location editing page.

### LINKED ISSUE #159: Implement Key Location page.

### PR DESCRIPTION
**Pull request by JuliusAlexanderIV**
closes #159

### CODE REVIEW

File: app/src/main/java/xyz/whereuat/whereuat/KeyLocationActivity.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    public void onCreate(Bundle saved_instance_state) {
+        super.onCreate(saved_instance_state);
+        setContentView(R.layout.activity_key_locs);
+
+        mMenu = (FloatingActionMenu) findViewById(R.id.menu);
+        mAdapter = new KeyLocationAdapter(this);
+        ((ListView) findViewById(R.id.key_locs_list)).setAdapter(mAdapter);
+
+        getSupportLoaderManager().initLoader(0, null, this);
+        initDrawer("Key Locations");
```
Comment (kangp3) [inline app/src/main/java/xyz/whereuat/whereuat/KeyLocationActivity.java]: Put into strings resource

File: app/src/main/java/xyz/whereuat/whereuat/KeyLocationActivity.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    /**
+     * This is called when the loader is first made. It is necessary to implement the LoaderManager
+     * callbacks.
+     *
+     * @param id the id of the loader, which is not needed because there is only one
+     * @param args miscellaneous arguments to help make the loader, of which there are none
+     * @return returns a new CursorLoader
+     */
+    public Loader<Cursor> onCreateLoader(int id, Bundle args) {
+        return new CursorLoader(this, null, KeyLocationEntry.COLUMNS, null, null, null) {
```
Comment (kangp3) [inline app/src/main/java/xyz/whereuat/whereuat/KeyLocationActivity.java]: Can pass `null` here https://developer.android.com/reference/android/content/ContentResolver.html#query(android.net.Uri, java.lang.String[], java.lang.String, java.lang.String[], java.lang.String)

File: app/src/main/java/xyz/whereuat/whereuat/ui/adapters/KeyLocationAdapter.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            }
+        });
+    }
+
+    private void handleWithEditKeyLocDialog(int id) {
+        FragmentManager fm = mActivity.getFragmentManager();
+        EditKeyLocationFragment add_contact_dialog = new EditKeyLocationFragment();
+        Bundle args = new Bundle();
+        args.putInt(Constants.KEY_LOC_ID_BUNDLE_EXTRA, id);
+        add_contact_dialog.setArguments(args);
```
Comment (kangp3) [inline app/src/main/java/xyz/whereuat/whereuat/ui/adapters/KeyLocationAdapter.java]: Could do the 88-91 functionality inside of a builder method. Did something similar with the key location confirmation dialog fragment.

File: app/src/main/java/xyz/whereuat/whereuat/ui/views/EditKeyLocationFragment.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    private void updateKeyLocName(final Activity activity, final int id, final String new_name) {
+        AsyncExecutor.service.submit(new Runnable() {
+            @Override
+            public void run() {
+                final int updated = KeyLocationUtils.buildUpdateNameByIdCommand(activity, id,
+                        new_name).call();
+                activity.runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        Log.d(TAG, Integer.toString(updated));
```
Comment (kangp3) [inline app/src/main/java/xyz/whereuat/whereuat/ui/views/EditKeyLocationFragment.java]: Unnecessary log?

File: app/src/main/java/xyz/whereuat/whereuat/KeyLocationActivity.java (hunk 1 above)
Comment (JuliusAlexanderIV) [inline app/src/main/java/xyz/whereuat/whereuat/KeyLocationActivity.java]: done

File: app/src/main/java/xyz/whereuat/whereuat/KeyLocationActivity.java (hunk 2 above)
Comment (JuliusAlexanderIV) [inline app/src/main/java/xyz/whereuat/whereuat/KeyLocationActivity.java]: done

File: app/src/main/java/xyz/whereuat/whereuat/ui/views/EditKeyLocationFragment.java (hunk 4 above)
Comment (JuliusAlexanderIV) [inline app/src/main/java/xyz/whereuat/whereuat/ui/views/EditKeyLocationFragment.java]: deleted

**Comment by kangp3** [comment]
lgtm

**Pull request merged**